packages/dev/core/src/Materials/meshDebugPluginMaterial.ts

```diff
@@ -0,0 +1,442 @@
+            uniformBuffer.updateFloat("m_uvScale", this._options.uvScale!);
+            uniformBuffer.updateVector3("m_uvColor1", this._options.uvColor1!);
+            uniformBuffer.updateVector3("m_uvColor2", this._options.uvColor2!);
+            uniformBuffer.updateVector3("m_materialColor", this._options.materialColor!);
+        }
+    }
+
+    /**
+     * Get shader code
+     * @param shaderType "vertex" or "fragment"
+     * @returns Shader code
+     */
+    public getCustomCode(shaderType: string): Nullable<{ [pointName: string]: string }> {
+        return shaderType === "vertex"
+            ? {
+                CUSTOM_VERTEX_DEFINITIONS: `
+                attribute float initialPass;
+
+                varying vec3 vBarycentric;
+                flat varying vec3 vVertexWorldPos;
+                flat varying float vPass;
+            `,
+                CUSTOM_VERTEX_MAIN_END: `
+                float vertexIndex = mod(float(gl_VertexID), 3.);
+
+                if (vertexIndex == 0.0) { 
+                    vBarycentric = vec3(1.,0.,0.); 
+                }
+                else if (vertexIndex == 1.0) { 
+                    vBarycentric = vec3(0.,1.,0.); 
+                }
+                else { 
+                    vBarycentric = vec3(0.,0.,1.); 
+                }
+
+                vVertexWorldPos = vPositionW;
+                vPass = initialPass;
+            `,
+              }
+            : {
+                CUSTOM_FRAGMENT_DEFINITIONS: `
+                #define PI 3.1415926535897932384626433832795
+
+                varying vec3 vBarycentric;
+                flat varying vec3 vVertexWorldPos;
+                flat varying float vPass;
+
+                vec3 m_applyShading(vec3 color) {
+                    vec3 N = vNormalW.xyz;
+                    vec3 L = normalize(vEyePosition.xyz - vPositionW.xyz);
+                    vec3 H = normalize(L + L);
+                    float LdotN = clamp(dot(L,N), 0., 1.);
+                    float HdotN = clamp(dot(H,N), 0., 1.);
+                    float specTerm = pow(HdotN, m_shadedSpecularPower);
+                    color *= vec3(1.) * (LdotN / PI);
+                    color += m_shadedSpecularColor * (specTerm / PI);
+                    return color;
+                }
+
+                #if MODE == 0 || MODE == 1
+                    float m_edgeFactor() {
+                        vec3 d = fwidth(vBarycentric);
+                        vec3 a3 = smoothstep(vec3(0.), d * m_wireframeThickness, vBarycentric);
+                        return min(min(a3.x, a3.y), a3.z);
+                    }
+                #endif
+
+                #if MODE == 1
+                    float m_cornerFactor() {
+                        vec3 worldPos = vPositionW;
+                        float dist = length(worldPos - vVertexWorldPos);
+                        float camDist = length(worldPos - vEyePosition.xyz);
+                        float d = sqrt(camDist) * 0.001;

```

Comment:
I'm not sure this `0.001` constant will work in all cases... In my testing, I had to change it depending on the fact I tested with a single cube or with the Alien head...

Maybe it should be an input of the shader?

Reply:
I'll make it an input, but I believe you could also achieve this same effect with inputs by changing the vertexRadius.